src/ResourceManager/Media/Commands.Media/MediaService/NewAzureRmMediaService.cs

```diff
@@ -0,0 +1,186 @@
+            HelpMessage = "The resource group name.")]
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = StorageAccountsParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = PrimaryStorageAccountParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The media service account name.")]
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = StorageAccountsParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The media service account name.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateLength(MediaServiceAccountNameMinLength, MediaServiceAccountNameMaxLength)]
+        [ValidatePattern(MediaServiceAccountNamePattern)]
+        [Alias(AccountNameAlias)]
+        public string AccountName { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = PrimaryStorageAccountParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The location.")]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = StorageAccountsParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The location.")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            ParameterSetName = PrimaryStorageAccountParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The primary storage account assosiated with the media account.")]
+        [Alias("Id")]
+        [ValidateNotNullOrEmpty]
+        public string StorageAccountId { get; set; }
+
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            ParameterSetName = StorageAccountsParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The storage accounts assosiated with the media account.")]
+        [ValidateNotNullOrEmpty]
+        public PSStorageAccount[] StorageAccounts { get; set; }
+
+        [Parameter(
+            Mandatory = false, 
+            HelpMessage = "The tags associated with the media service account.")]
+        [ValidateNotNull]
+        public Hashtable Tags { get; set; }
+
+        [Parameter(Mandatory = false,
+           HelpMessage = "Force to new a media service without confirm.")]
+        public SwitchParameter Force { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            try
```

Comment:
ShouldProcess should wrap this whole execution path.  A user specifying -WhatIf should not get an excption of the media service account already exists.